test/unit/user_test.rb

```diff
@@ -17,8 +17,18 @@ def setup
 
   test "should have mail when updating" do
     u = FactoryGirl.create(:user, :mail => nil)
-    u.firstname = 'Bob'
-    refute_valid u, :mail
+    as_user(u) do
+      u.firstname = 'Bob'
+      refute_valid u, :mail
+    end
+  end
+
+  test "mail is still opitonal when admin does the updating" do
+    as_admin do
+      u = FactoryGirl.create(:user, :mail => nil)
+      u.firstname = 'Bob'
+      assert_valid u
+    end
```

Comment:
I think you need to call `save!` to trigger something. Or use `update_attributes!`.